generator/booking_dataset.hpp

```diff
@@ -85,24 +87,26 @@ class BookingDataset
   bool CanBeBooking(FeatureBuilder1 const & fb) const;
 
   inline size_t Size() const { return m_hotels.size(); }
-  Hotel const & GetHotel(size_t index) const;
-  Hotel & GetHotel(size_t index);
-  vector<size_t> GetNearestHotels(ms::LatLon const & latLon, size_t limit,
+  Hotel const & GetHotelById(uint32_t id) const;
+  Hotel & GetHotelById(uint32_t id);
+  vector<uint32_t> GetNearestHotels(ms::LatLon const & latLon, size_t limit,
                                   double maxDistance = 0.0) const;
   bool MatchByName(string const & osmName, vector<size_t> const & bookingIndexes) const;
 
-  void BuildHotel(size_t hotelIndex, function<void(FeatureBuilder1 &)> const & fn) const;
+  void BuildHotels(function<void(FeatureBuilder1 &)> const & fn) const;
 
 protected:
-  vector<Hotel> m_hotels;
+  map<uint32_t, Hotel> m_hotels;
 
   using TPoint = boost::geometry::model::point<float, 2, boost::geometry::cs::cartesian>;
   using TBox = boost::geometry::model::box<TPoint>;
-  using TValue = pair<TBox, size_t>;
+  using TValue = pair<TBox, uint32_t>;
 
   // Create the rtree using default constructor.
   boost::geometry::index::rtree<TValue, boost::geometry::index::quadratic<16>> m_rtree;
 
+  void BuildHotel(Hotel const & hotel, function<void(FeatureBuilder1 &)> const & fn) const;
```

Comment:
I don't think it's necessary here. They all serve the same purpose and are not used outside the class.